code/modules/mob/living/basic/jungle/seedling/seedling.dm

```diff
@@ -0,0 +1,314 @@
+	ai_controller.set_blackboard_key(BB_RAPIDSEEDS_ABILITY, seed_attack)
+	var/datum/action/cooldown/mob_cooldown/solarbeam/beam_attack = new(src)
+	beam_attack.Grant(src)
+	ai_controller.set_blackboard_key(BB_SOLARBEAM_ABILITY, beam_attack)
+
+	var/petal_color = pick(possible_colors)
+
+	petal_neutral = mutable_appearance(icon, "[icon_state]_overlay")
+	petal_neutral.color = petal_color
+
+	petal_warmup = mutable_appearance(icon, "[icon_state]_charging_overlay")
+	petal_warmup.color = petal_color
+
+	petal_active = mutable_appearance(icon, "[icon_state]_fire_overlay")
+	petal_active.color = petal_color
+
+	petal_dead = mutable_appearance(icon, "[icon_state]_dead_overlay")
+	petal_dead.color = petal_color
+
+	AddElement(/datum/element/wall_smasher)
+	AddComponent(/datum/component/obeys_commands, seedling_commands)
+	RegisterSignal(src, COMSIG_HOSTILE_PRE_ATTACKINGTARGET, PROC_REF(pre_attack))
+	if(tamer)
+		befriend(tamer)
+	update_appearance()
+
+/mob/living/basic/seedling/proc/pre_attack(mob/living/puncher, atom/target)
+	SIGNAL_HANDLER
+
+	if(istype(target, /obj/machinery/hydroponics))
+		treat_hydro_tray(target)
+		return COMPONENT_HOSTILE_NO_ATTACK
+
+	if(!held_can)
+		return
+
+	if(istype(target, /obj/structure/sink) || istype(target, /obj/structure/reagent_dispensers))
+		INVOKE_ASYNC(held_can, TYPE_PROC_REF(/obj/item, melee_attack_chain), src, target)
+		return COMPONENT_HOSTILE_NO_ATTACK
+
+
+///seedlings can water trays, remove weeds, or remove dead plants
+/mob/living/basic/seedling/proc/treat_hydro_tray(obj/machinery/hydroponics/hydro)
+
+	if(hydro.plant_status == HYDROTRAY_PLANT_DEAD)
+		src.balloon_alert(src, "dead plant removed!")
+		hydro.set_seed(null)
+
+	if(hydro.weedlevel > 0)
+		src.balloon_alert(src, "weeds uprooted!")
+		hydro.set_weedlevel(0)
+		return
+
+	var/list/can_reagents = held_can?.reagents.reagent_list
+
+	if(!length(can_reagents))
+		return
+
+	if((locate(/datum/reagent/water) in can_reagents) && (hydro.waterlevel < hydro.maxwater))
+		INVOKE_ASYNC(held_can, TYPE_PROC_REF(/obj/item, melee_attack_chain), src, hydro)
+		return
+
+/mob/living/basic/seedling/UnarmedAttack(atom/attack_target, proximity_flag, list/modifiers)
+	. = ..()
+
+	if(!. || !proximity_flag || held_can)
+		return
+
+	if(!istype(attack_target, /obj/item/reagent_containers/cup/watering_can))
+		return
+
+	var/obj/item/can_target = attack_target
+	can_target.forceMove(src)
+	held_can = can_target
+	update_appearance()
+
+/mob/living/basic/seedling/proc/change_combatant_state(state)
+	combatant_state = state
+	update_appearance()
+
+/mob/living/basic/seedling/update_overlays()
+	. = ..()
+	if(stat == DEAD)
+		. += petal_dead
+		return
+
+	switch(combatant_state)
+		if(SEEDLING_STATE_NEUTRAL)
+			. += petal_neutral
+		if(SEEDLING_STATE_WARMUP)
+			. += petal_warmup
+		if(SEEDLING_STATE_ACTIVE)
+			. += petal_active
+
+	if(combatant_state == SEEDLING_STATE_NEUTRAL && held_can)
+		. +=  mutable_appearance(icon, "seedling_can_overlay")
```

Comment:
You already check for combatant state via the switch, you can just throw on the can overlay there
```suggestion
	switch(combatant_state)
		if(SEEDLING_STATE_NEUTRAL)
			. += petal_neutral
			if(held_can)
				. += mutable_appearance(icon, "seedling_can_overlay")
		if(SEEDLING_STATE_WARMUP)
			. += petal_warmup
		if(SEEDLING_STATE_ACTIVE)
			. += petal_active

```